File: 09.Greed贪心/leetcode_3016_minimumPushes.cpp
#include <iostream>
#include <string>
#include <algorithm>
#include <vector>
using namespace std;

class Solution {
public:
    int minimumPushes(string word) {
        vector<int> cnt(26, 0);
        for (auto& c : word){
            cnt[c-'a']++;
        }
        sort(cnt.begin(), cnt.end(), greater<int>());
        int res = 0;
        for (int i=0; i<26; i++){
            res += cnt[i]*(i/8+1);
        }
        return res;
    }
};

int main(){
    Solution s;
    string word = "abcde";
    cout << s.minimumPushes(word) << endl;
}

File: 09.Greed贪心/leetcode_3016_minimumPushes.py
from collections import Counter

class Solution:
    def minimumPushes(self, word: str) -> int:
        cnt = Counter(word)
        val = sorted(cnt.values(), reverse = True)
        return sum(val[i]*(i//8+1) for i in range(len(val)))

if __name__ == "__main__":
    s = Solution()
    word = "abcde"
    print(s.minimumPushes(word))
